plugin/src/main/kotlin/me/tylerbwong/gradle/metalava/task/MetalavaSignature.kt

```diff
@@ -18,17 +51,27 @@ internal object MetalavaSignature : MetalavaTaskContainer() {
     ): TaskProvider<JavaExec> {
         return with(project) {
             tasks.register(name, JavaExec::class.java) {
+                require(extension.sourcePaths.isNotEmpty()) {"sourcePaths cannot be empty."}
+
                 group = "documentation"
                 this.description = description
                 mainClass.set("com.android.tools.metalava.Driver")
                 classpath(extension.metalavaJarPath?.let { files(it) } ?: getMetalavaClasspath(extension.version))
-                val sources = file("src")
-                    .walk()
-                    .maxDepth(2)
-                    .onEnter { !it.name.contains("test", ignoreCase = true) }
-                    .filter { it.isDirectory && (it.name == "java" || it.name == "kotlin") }
-                    .toList()
-                inputs.files(sources)
+
+                val directoryCompareFunction = if (extension.ignoreSourcePathsExactMatch) isDirectoryToBeIgnoredStrict else isDirectoryToBeIgnored
```

Comment:
Perhaps there's something clever wtih partial application, or somesuch, that one could do (ie the difference being equals() vs contains()). I didn't want to spend a significant amount of time wrestling with that, however (and I'm not entirely sure that it'd add much to the code, other than making it harder to read).